lib/experimental/fonts/font-library/class-wp-rest-font-families-controller.php

```diff
@@ -19,388 +17,512 @@
-		$this->namespace = 'wp/v2';
-		$this->post_type = 'wp_font_family';
-	}
+	protected $allow_batch = false;
 
 	/**
-	 * Registers the routes for the objects of the controller.
+	 * Checks if a given request has access to font families.
 	 *
 	 * @since 6.5.0
+	 *
+	 * @param WP_REST_Request $request Full details about the request.
+	 * @return true|WP_Error True if the request has read access, WP_Error object otherwise.
 	 */
-	public function register_routes() {
-		register_rest_route(
-			$this->namespace,
-			'/' . $this->rest_base,
-			array(
-				array(
-					'methods'             => WP_REST_Server::READABLE,
-					'callback'            => array( $this, 'get_items' ),
-					'permission_callback' => array( $this, 'update_font_library_permissions_check' ),
-				),
-			)
-		);
+	public function get_items_permissions_check( $request ) { // phpcs:ignore VariableAnalysis.CodeAnalysis.VariableAnalysis.UnusedVariable -- required by parent class
+		$post_type = get_post_type_object( $this->post_type );
 
-		register_rest_route(
-			$this->namespace,
-			'/' . $this->rest_base,
-			array(
-				array(
-					'methods'             => WP_REST_Server::EDITABLE,
-					'callback'            => array( $this, 'install_fonts' ),
-					'permission_callback' => array( $this, 'update_font_library_permissions_check' ),
-					'args'                => array(
-						'font_family_settings' => array(
-							'required'          => true,
-							'type'              => 'string',
-							'validate_callback' => array( $this, 'validate_install_font_families' ),
-						),
-					),
-				),
-			)
-		);
+		if ( ! current_user_can( $post_type->cap->read ) ) {
+			return new WP_Error(
+				'rest_cannot_read',
+				__( 'Sorry, you are not allowed to access font families.', 'gutenberg' ),
+				array( 'status' => rest_authorization_required_code() )
+			);
+		}
 
-		register_rest_route(
-			$this->namespace,
-			'/' . $this->rest_base,
-			array(
-				array(
-					'methods'             => WP_REST_Server::DELETABLE,
-					'callback'            => array( $this, 'uninstall_fonts' ),
-					'permission_callback' => array( $this, 'update_font_library_permissions_check' ),
-					'args'                => $this->uninstall_schema(),
-				),
-			)
-		);
+		return true;
 	}
 
 	/**
-	 * Returns validation errors in font families data for installation.
+	 * Checks if a given request has access to a font family.
 	 *
 	 * @since 6.5.0
 	 *
-	 * @param array[] $font_families Font families to install.
-	 * @param array   $files         Files to install.
-	 * @return array $error_messages Array of error messages.
+	 * @param WP_REST_Request $request Full details about the request.
+	 * @return true|WP_Error True if the request has read access, WP_Error object otherwise.
 	 */
-	private function get_validation_errors( $font_family_settings, $files ) {
-		$error_messages = array();
+	public function get_item_permissions_check( $request ) {
+		return $this->get_items_permissions_check( $request );
```

Comment:
As noted else, this permission check is not id based.

Reply:
Updated here: https://github.com/WordPress/gutenberg/pull/58333